### PR TITLE
Add periodic flush count to flush stats

### PR DESCRIPTION
**Pull request by dnhatn**
Currently, a flush stats contains only the total flush which is the sum
of manual flush (via API) and periodic flush (async triggered when the
uncommitted translog size is exceeded the flush threshold). Sometimes,
it's useful to know these two numbers independently. This commit tracks
and returns the periodic flush count in a flush stats.

Relates #29125

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/flush/FlushStats.java [hunk 1]
```diff
@@ -85,6 +96,7 @@ public TimeValue getTotalTime() {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(Fields.FLUSH);
         builder.field(Fields.TOTAL, total);
+        builder.field(Fields.PERIODIC, periodic);
```
Comment (s1monw) [inline server/src/main/java/org/elasticsearch/index/flush/FlushStats.java]: can we have a rest test that checks that this is there?

**Review by s1monw — COMMENTED** [review]
left one comment

File: server/src/main/java/org/elasticsearch/index/flush/FlushStats.java (hunk 1 above)
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/index/flush/FlushStats.java]: @s1monw 
I had a test but I could not verify the periodic value as the periodic flush is executed async. Do you have any suggestion for this? Or just check its presence (eg. 0 is ok)?

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/index/flush/FlushStats.java]: I used the analogue of `gt: { periodic: 0}` when I did something similar.

Comment (s1monw) [reply server/src/main/java/org/elasticsearch/index/flush/FlushStats.java]: yeah ++

Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/index/flush/FlushStats.java]: I will go this suggestion. Thanks @DaveCTurner and @s1monw.

**Comment by dnhatn** [comment]
@s1monw I've added a rest test. Can you please have another look? Thank you!

**Review by s1monw — APPROVED** [review]
LGTM

**Comment by dnhatn** [comment]
Thanks @s1monw and @DaveCTurner.

**Pull request merged**